core/src/mindustry/editor/WaveInfoDialog.java

```diff
@@ -165,20 +204,29 @@ void buildGroups(){
         table.margin(10f);
 
         if(groups != null){
-            groups.sort(g -> g.begin);
+            if(sort == Sort.begin) groups.sort(g -> g.begin * (reverseSort ? -1 : 1));
```

Comment:
Instead of a bunch of if statements, SortMode should have a final `Floatf<UnitGroup>` field to specify order. No negations are necessary, just call `groups.reverse()` after sorting if `reverseSort` is true.